pkg/testutil/testutil.go

```diff
@@ -17,15 +17,33 @@ package testutil
 import (
 	"net/url"
 	"runtime"
+	"strings"
 	"testing"
+	"time"
 )
 
-// WARNING: This is a hack.
-// Remove this when we are able to block/check the status of the go-routines.
-func ForceGosched() {
-	// possibility enough to sched up to 10 go routines.
-	for i := 0; i < 10000; i++ {
-		runtime.Gosched()
+func runnableGoroutineNumber() int {
```

Comment:
without a comment, i have no idea how this works (why split by \n\n, etc.).

and i am wondering if there is a simpler way.